internal/nprintf/nprintf.go

```diff
@@ -0,0 +1,23 @@
+package nprintf
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// NPrintf is a helper function to replace placeholders in a string with values from a map
+func NPrintf(format string, params map[string]interface{}) string {
```

Comment:
IMO it would be preferable to leave this in `equinix/` and deprecate it.  While there is some reduction of repetition here compared to using `fmt.Sprintf`, the repetition has the benefit of being explicit and flexible.

Reply:
The nprintf extraction was made to simplify the changes and review needed for https://github.com/equinix/terraform-provider-equinix/pull/622 where additional packages and functions were being moved out of `equinix/` as part of #106 .

Reply:
I think in this case I'd usually lean towards duplicating (not moving) nprintf into a separate package and marking the NPrintf function as deprecated in the same PR that moves the NE resources.  Duplicating the function enables us to move only the NE resources without touching anything else, and doing that in one PR makes it clear that this is tech debt we're taking on to accomplish a specific goal.

Given our CODEOWNERS configuration, though, I could see having a separate PR to introduce the nprintf package; IMO a PR like that should only touch resources for which we are the owners, so we would introduce the nprintf package as a deprecated approach that exists only to facilitate code migration and update the ECX resources--without moving them, since they're going away soon--to demonstrate that nprintf package is a drop-in replacement for the private nprintf function.  A follow-up PR could move the NE resources and it would be up to the NE team to review and adopt that PR.

Reply:
Duplicated and deprecated. (removed: 6490d46afdac4feedd0cb0975771145154f76ee7 and e7708e9175aa5ca3bae2abbb7586d287bb0168ae, the first may be cherry-picked later)